Sources/Helpers/OADiscountHelper.h

```diff
@@ -53,5 +53,6 @@
 + (OADiscountHelper *)instance;
 
 - (void) checkAndDisplay;
+- (BOOL) isVisible;
```

Comment:
Drop it. DiscountBanner is a ToolbarViewController. It seems you should check for any toolbar visibility.